physical/s3/s3.go

```diff
@@ -188,16 +188,17 @@ func (s *S3Backend) Get(ctx context.Context, key string) (*physical.Entry, error
 	if resp == nil {
 		return nil, fmt.Errorf("got nil response from S3 but no error")
 	}
+	defer resp.Body.Close()
 
-	data := make([]byte, *resp.ContentLength)
-	_, err = io.ReadFull(resp.Body, data)
+	data := bytes.NewBuffer(nil)
```

Comment:
To retain the more efficient pre-allocation when ContentLength is known (presumably the normal case), consider:
```
	var cap int64
	if resp.ContentLength != nil {
		cap = *resp.ContentLength
	}

	data := bytes.NewBuffer(make([]byte, 0, cap))
```

Reply:
I looked at that as a "efficiency doesn't really matter because network aspects will dominate the total time" but I guess if it's a 20mb object it actually could, so I'll do that.